mockspec/rules

```diff
@@ -7,6 +7,16 @@ k8s; pkg/k8s/mock_controller_generated.go; github.com/openservicemesh/osm/pkg/k8
 
 # pkg/debugger
 debugger; pkg/debugger/mock_debugger_generated.go; github.com/openservicemesh/osm/pkg/debugger; XDSDebugger
+debugger; pkg/debugger/mock_debugger_infra_client_generated.go; github.com/openservicemesh/osm/pkg/debugger; DebuggerInfraClient
```

Comment:
Instead of creating generated mocks for all of these, can we manually create a struct that adheres to the interface to support what we want for tests? This is likely to be much more idiomatic for the smaller interfaces. Debugger is a good one to start with